src/ui/views/tui_textfield.rs

```diff
@@ -313,24 +313,26 @@ fn autocomplete(
     if let Some(ref mut ct) = completion_tracker {
         ct.index = if reversed {
             ct.index.checked_sub(1).unwrap_or(ct.candidates.len() - 1)
-        } else if ct.index + 1 < ct.candidates.len() {
-            ct.index + 1
         } else {
-            ct.index
+            (ct.index + 1) % ct.candidates.len()
```

Comment:
I prefer the autocomplete suggestions does not loop itself.
If a user would like to go back on a suggestion, one can use reverse order or restart the autocomplete by typing something.